dist/IO/lib/IO/Select.pm

```diff
@@ -57,7 +57,21 @@ sub _fileno
  my($self, $f) = @_;
  return unless defined $f;
  $f = $f->[0] if ref($f) eq 'ARRAY';
- ($f =~ /^\d+$/) ? $f : fileno($f);
+ if($f =~ /^\d+$/) { # plain file number
```

Comment:
Shouldn't this be `/^[0-9]$/`

Reply:
Hi! You likely meant `/^[0-9]+$/` (with the `+`) and you are right: https://perl-begin.org/tutorials/bad-elements/#using-character-classes-with-unicode-text .

Reply:
Yep. Fixed and re-smoking.